pkg/clients/postgres/clients.go

```diff
@@ -0,0 +1,472 @@
+package postgres
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/jackc/pgtype"
+	"github.com/mainflux/mainflux/internal/postgres"
+	"github.com/mainflux/mainflux/pkg/clients"
+	"github.com/mainflux/mainflux/pkg/errors"
+	"github.com/mainflux/mainflux/pkg/groups"
+)
+
+type ClientRepository struct {
+	DB postgres.Database
+}
+
+func (repo ClientRepository) Update(ctx context.Context, client clients.Client) (clients.Client, error) {
+	var query []string
+	var upq string
+	if client.Name != "" {
+		query = append(query, "name = :name,")
+	}
+	if client.Metadata != nil {
+		query = append(query, "metadata = :metadata,")
+	}
+	if len(query) > 0 {
+		upq = strings.Join(query, " ")
+	}
+	client.Status = clients.EnabledStatus
+	q := fmt.Sprintf(`UPDATE clients SET %s updated_at = :updated_at, updated_by = :updated_by
+        WHERE id = :id AND status = :status
+        RETURNING id, name, tags, identity, secret,  metadata, COALESCE(owner_id, '') AS owner_id, status, created_at, updated_at, updated_by`,
+		upq)
+
+	return repo.update(ctx, client, q)
+}
+
+func (repo ClientRepository) UpdateTags(ctx context.Context, client clients.Client) (clients.Client, error) {
+	client.Status = clients.EnabledStatus
+	q := `UPDATE clients SET tags = :tags, updated_at = :updated_at, updated_by = :updated_by
+        WHERE id = :id AND status = :status
+        RETURNING id, name, tags, identity, metadata, COALESCE(owner_id, '') AS owner_id, status, created_at, updated_at, updated_by`
+
+	return repo.update(ctx, client, q)
+}
+
+func (repo ClientRepository) UpdateIdentity(ctx context.Context, client clients.Client) (clients.Client, error) {
+	q := `UPDATE clients SET identity = :identity, updated_at = :updated_at, updated_by = :updated_by
+        WHERE id = :id AND status = :status
+        RETURNING id, name, tags, identity, metadata, COALESCE(owner_id, '') AS owner_id, status, created_at, updated_at, updated_by`
+
+	return repo.update(ctx, client, q)
+}
+
+func (repo ClientRepository) UpdateSecret(ctx context.Context, client clients.Client) (clients.Client, error) {
+	q := `UPDATE clients SET secret = :secret, updated_at = :updated_at, updated_by = :updated_by
+        WHERE identity = :identity AND status = :status
```

Comment:
```suggestion
        WHERE id = :id AND status = :status
```

Reply:
Also for things we return the `secret` while for users not necessarily after updating thing secret

Reply:
for unification reasons, I could return secret for both